src/utils/useStateReadState/index.ts

```diff
@@ -1,123 +1,45 @@
-  Signers: T[]
-  StartEpoch: number
-  UnlockDuration: number
-}
-
-const emptyActorState: LotusRPCActorState<any> = {
-  Balance: '0',
-  Code: { '/': '' },
-  State: null
-}
-
-/////// implemented until https://github.com/glifio/graph/issues/33
-export const useStateReadStateQuery = <T = any>(
-  baseOptions: QueryHookOptions<
-    T,
-    {
-      address: string
-    }
-  >
-): { data: LotusRPCActorState<T>; error: Error; loading: boolean } => {
-  const { lotusApi, networkName } = useEnvironment()
-  const [actorState, setActorState] = useState<LotusRPCActorState<T> | null>(
-    null
-  )
-  const [loading, setLoading] = useState(true)
-  const [error, setError] = useState<Error>(undefined)
-  const [fetchedFor, setFetchedFor] = useState<string>('')
-
-  const apolloClient = useApolloClient()
+export const useStateReadState = <T = object | null>(
+  address: string
+): UseStateReadStateResult<T> => {
+  const [data, setData] = useState<LotusActorState<T> | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [notFound, setNotFound] = useState<boolean>(false)
+  const [error, setError] = useState<Error | null>(null)
+  const { lotusApi } = useEnvironment()
 
   useEffect(() => {
-    const fetchState = async () => {
-      try {
-        const res = await lotusApi.request<LotusRPCActorState<any>>(
-          'StateReadState',
-          baseOptions.variables.address,
-          null
-        )
-
-        if (res && decodeActorCID(res.Code, networkName).includes('multisig')) {
-          res.State = res.State as MsigState<string>
-          const signers = await Promise.all(
-            res.State.Signers.map(async (id: string) => {
-              const { data } = await apolloClient.query<AddressQuery>({
-                query: AddressDocument,
-                variables: {
-                  address: id
-                }
-              })
-
-              return { id, robust: data?.address?.robust || '' } as Address
-            })
-          )
-
-          const availableBalance = await lotusApi.request<string>(
-            'MsigGetAvailableBalance',
-            baseOptions.variables.address,
-            null
-          )
-
-          setActorState({
-            ...res,
-            State: {
-              ...res.State,
-              Signers: signers,
-              AvailableBalance: availableBalance
-            }
-          })
-        } else if (res) {
-          setActorState({ ...res })
-        }
-      } catch (err) {
-        if (err instanceof Error) {
-          if (err.message.includes('actor not found')) {
-            setActorState({ ...emptyActorState })
-          } else setError(err)
-        } else setError(new Error('Failed to fetch'))
-      } finally {
-        setLoading(false)
-      }
-    }
-
-    const firstLoad = !actorState && loading && !error
-    const newLoad = fetchedFor !== baseOptions?.variables?.address && !error
-
-    if (firstLoad || newLoad) {
```

Comment:
This logic is removed now, but I wonder if we really need it. I can reimplement something similar if you think we do